### ISSUE TITLE
HW-Integration: Default values

### ISSUE THREAD

**Issue by eichelbe** [QualiMaster/qm-issues#19]
- Define default parameters (for hardware) as values in the configuration - shall be possible right now.
- set default parameters before switching an algorithm... at least for hardware, check with sw-algorithm whether this is also needed

**Comment by cuiqin** [comment]
but there is no configured parameters. Which parameter needs for hardware?

**Comment by eichelbe** [comment]
Nick told me about window size, which is currently passed in via the Storm config object / Map.

**Comment by npav** [comment]
It is not a hardware parameter rather than a Hayashi-Yoshida parameter. The reason we discussed about it is because the hardware needs to know the algorithm parameters after switch. Currently, windowSize is the only parameter for the algorithm (Cui I also told you about windowAdvance, but let's leave that one out for now as it is a little obscure and we need to talk about it more in depth) and it is defined in the Topology file as a config entry.

**Comment by cuiqin** [comment]
Fine:) Shall I add the windowAdvance parameter to the correlation family and then you can adjust your implementation for that? Afterwards, I also need to modify the pipeline instantiation script in order to set that parameter before we run it.

**Comment by npav** [comment]
Just let the windowAdvance a constant of 1 second as it is now.

**Comment by cuiqin** [comment]
Oh..yes, actually I wanted to add the windowSize parameter;). Shall I add it for the correlation family?

**Comment by eichelbe** [comment]
No algorithm parameter without family parameter ;)

**Comment by cuiqin** [comment]
True. We need to reconfigure the correlation family in this case.

**Comment by npav** [comment]
Yes, we should add the windowSize as a correlation family parameter. Though I have a semantics-related concern as there might be non window-based correlation algorithms in the future.

**Comment by eichelbe** [comment]
So far, algorithms are free to ignore individual parameters. The profiles to be derived from algorithms shall then indicate that the parameter has no effect for the specific algorithm, so varying it at runtime is not needed.

**Comment by cuiqin** [comment]
The windowSize parameter has been included for correlation families and a default value 30s for this parameter has been set for all correlation-related algorithms in the PriorityPipeline. @npav could you please adjust the correlation-related implementations?

**Comment by eichelbe** [comment]
Default values for parameters are on the way, in the model and also in QM-IConf. Instantiation/Adaptation will follow.

**Comment by cuiqin** [comment]
The default values for parameters are in the model now. The changes on the instantiation as well as QM-IConf have been committed.

**Comment by eichelbe** [comment]
Make the adaptation ready for parameters.

**Comment by ekateriniioannou** [comment]
Cui needs default values for parameters from TSI and L3S

**Comment by npav** [comment]
@cuiqin We have included the necessary changes for the parameter change of the windowSize (the only parameter in correlation at the moment). We tested locally to the point that was possible and it seems fine. We also uploaded the Jenkinss build to the QM cluster and it also seems to be working fine, but we cannot send a parameter change signal, so can you please replicate the behaviour and test if everything is working fine?

**Comment by cuiqin** [comment]
@npav Your can send a test parameter through the infrastructure using the command 
./cli.sh setParam <pipeline> <pipelineElement> <param> <value>

**Comment by cuiqin** [comment]
It seems to me here it doesn't show the complete command by the text with the angular brackets. Please find the commands in: https://www.l3s.de/wiki/doku.php?id=qualimaster:qualimaster:infrastructure-install&#infrastructure_scripts

**Comment by eichelbe** [comment]
Adaptation model adjusted for default parameters. Cannot commit by now, as QM-IConf needs a fix release. Handed over to coordination layer as command set to allow reordering. Defaults are sent by adaptation only once and CoordinationLayer shall cache the parameters so that sending them along with algorithm switch (if required also only for hardware) becomes possible. 

This needs changes in the CoordinationLayer.

**Comment by eichelbe** [comment]
Signals can now transport multiple parameter changes.
Coordination layer caches actual parameters, combines multiple commands if needed and passes them on.

**Comment by eichelbe** [comment]
Pipeline needs now to consider multiple parameters... getting rid of now deprecated methods ;)

**Comment by eichelbe** [comment]
-> Cui for generated pipelines

**Comment by cuiqin** [comment]
The needed modification for generated pipelines is done.
@ap0n please update the infrastructure jars in your cluster as well as the needed modification for your algorithms(similar implementation of the notifyParameterChange method as the generated pipeline).
